src/Microsoft.FeatureManagement/FeatureManagementError.cs

```diff
@@ -16,6 +16,46 @@ public enum FeatureManagementError
         /// <summary>
         /// A feature filter configured for the feature being evaluated is an ambiguous reference to multiple registered feature filters.
         /// </summary>
-        AmbiguousFeatureFilter
+        AmbiguousFeatureFilter,
+
+        /// <summary>
+        /// A feature variant assigner that was listed for variant assignment was not found.
+        /// </summary>
+        MissingFeatureVariantAssigner,
+
+        /// <summary>
+        /// The feature variant assigner configured for the feature being evaluated is an ambiguous reference to multiple registered feature variant assigners.
+        /// </summary>
+        AmbiguousFeatureVariantAssigner,
+
+        /// <summary>
+        /// An assigned feature variant does not have a valid configuration reference.
+        /// </summary>
+        MissingConfigurationReference,
+
+        /// <summary>
+        /// A feature variant assigner being used in feature evaluation is invalid.
+        /// </summary>
+        InvalidFeatureVariantAssigner,
+
+        /// <summary>
+        /// A feature that was requested for evaluation was not found.
+        /// </summary>
+        MissingFeature,
+
+        /// <summary>
+        /// A dynamic feature does not have any feature variants registered.
+        /// </summary>
+        MissingVariants,
```

Comment:
>MissingVariants [](http://example.com/codeflow?start=8&length=15)

Sometimes we call it "FeatureVariant", for example in `MissingFeatureVariantAssigner` and sometimes just "Variant" like here `MissingVariants` (btw, should it have "s"?). Do we want to be consistent and use "FeatureVariant" everywhere?

Reply:
Thanks, I'll update to use "FeatureVariant" for consistency as you mentioned.

I think removing the "s" is a good idea since having a single feature variant listed is valid.